### PR TITLE
support extendr_polars

### PR DESCRIPTION
**Pull request by sorhawell**
This PR will add support for [`extendr_polars`](https://github.com/rpolars/extendr_polars) which allows rust extensions to polars in R. [Examples](https://github.com/rpolars/extendrpolarsexamples)

### CODE REVIEW

**Comment by sorhawell** [comment]
this update is needed  to support the crate [extendr_polars](https://github.com/rpolars/extendr_polars) 

`extendr_polars` crate allows an R package developer to interface with the polars R package via  arrow-import/export r-polars of DataFrame. This is zero cost and much safer than transmuting the inner rust-polars DataFrame between compilation units and differing versions.

an example R package [is found here](https://github.com/rpolars/extendrpolarsexamples)

**Review by etiennebacher — APPROVED** [review]
@sorhawell I can approve if you want but I have no clue if this is good or if there are mistake. I suppose it works fine since you made some tests repos using it, but maybe @eitsupi would be more comfortable reviewing this?

File: src/rust/src/arrow_interop/to_rust.rs [hunk 1]
```diff
@@ -160,3 +134,63 @@ pub fn to_rust_df(rb: Robj) -> Result<pl::DataFrame, String> {
...
+    let us = crate::utils::robj_str_ptr_to_usize(&robj_str)?;
+    let boxed_stream = unsafe { Box::from_raw(us as *mut ffi::ArrowArrayStream) };
+
+    //consume stream and produce a r-polars Series return as Robj
+    let s = consume_arrow_stream_to_s(boxed_stream)?;
+    Ok(s)
+}
+
+// implementation of consuming stream to Series. Stream is drop/released hereafter.
+fn consume_arrow_stream_to_s(boxed_stream: Box<ffi::ArrowArrayStream>) -> RResult<pl::Series> {
```
Comment (eitsupi) [inline src/rust/src/arrow_interop/to_rust.rs:158]: `to_s` may be a little ambiguous?

**Review by eitsupi — COMMENTED** [review]
Sorry, I don't have enough knowledge to review this.......

File: src/rust/src/arrow_interop/to_rust.rs (hunk 1 above)
Comment (sorhawell) [reply src/rust/src/arrow_interop/to_rust.rs]: thx I have renamed  to `...to_series`

**Comment by sorhawell** [comment]
@etiennebacher  arrow-protocol is really hairy ^^ I have looked in the sparse official arrow docs for days to get an ok understanding. 

I have added a round trip unit test with [b38f4d8](https://github.com/pola-rs/r-polars/pull/326/commits/b38f4d8194fdc793aa02080a2c9b6f2853c49c40) and added some notes for what is going on and when to use this.

**Comment by eitsupi** [comment]
Thanks!

**Pull request merged**